### ISSUE TITLE
Provide antialias / multisampling settings for the 3d view

### ISSUE THREAD

**Issue by dronus** [openscad/openscad#1769]
While it can be enforced by some video card settings, for some cards it can't, so it would be cool if OpenSCAD's settings provide a mean to enable polygon antialiasing or multisampling in the 3d view. 

I do not have a complete picture, but it may be true that the screen space operations needed for preview display of CGAL operations do not allow for native multisampling, in this case rendering to a larger off-screen buffer could be used. This needs to be provided by OpenSCAD as it can't be enforced by video card settings.

**Comment by t-paul** [comment]
There is indeed a comment that might be related at https://github.com/openscad/openscad/blob/master/src/openscad.cc#L700
```
// turn on anti-aliasing
disabled by clifford wolf: adds rendering artefacts with OpenCSG
```
[Updating the OpenGL render code](https://github.com/openscad/openscad/wiki/Project:-Improve-OpenGL-rendering) is a big topic on the wishlist, this could be reviewed in scope of that once someone is working on that.

**Comment by dronus** [comment]
I thought that the "preview" mode uses depth buffer mechanisms for CSG bool operations. Most 3rd party scenegraph frameworks won't support those tricks properly I guess, so the OpenGL code could maybe not rely on this, or the preview feature would be dropped I fear.
On the other hand, I experience full CGAL computation even for preview quite often on newer versions. I will look into this.

**Comment by t-paul** [comment]
CGAL render while doing preview would be caused by:
- `render()` (explicitely forces CGAL)
- `minkowski()` (not supported by OpenCSG)
- `hull()` (not supported by OpenCSG)

(That what I currently know about, we probably need to check if there are more triggers)

**Comment by MichaelAtOz** [comment]
- resize()

**Comment by dronus** [comment]
Yeah, that's it. My stupid. Preview works well on designs not using these features.

**Comment by bebr42** [comment]
Maybe I don't understand the solution. 
But on my (Windows 7) machine the preview is quite coarse and the edges use no antialiasing at all. 
Would be nice to fix this.

**Comment by nophead** [comment]
On my last computer with an Nvidia card I could turn on antialiasing for
OpenSCAD. Now I have an Intel graphics chip the only option seems to be to
disable it or leave it to the application to decide.

**Comment by bebr42** [comment]
Exactly, same situation here, Lenovo Thinkpad, which delivers otherwise nice performance, even on larger external monitors resolutions.
So we want to be able to control the 3D settings from within openScad.

**Comment by stefano2734** [comment]
After 9 years it can be solved. 
Related issues by openGL and openCSG here are solved in year 2020.
So release 2021 is perhaps solved and also  actual dev.
